File: app/src/main/java/com/gmail/fattazzo/formula1world/domain/F1Constructor.java
package com.gmail.fattazzo.formula1world.domain;

import java.io.Serializable;

/**
 * @author fattazzo
 *         <p/>
 *         date: 05/06/17
 */
public class F1Constructor implements Serializable {

    public String constructorRef;

    public String url;

    public String name;

    public String nationality;

    @Override
    public String toString() {
        return "Constructor{" +
                "constructorRef='" + constructorRef + '\'' +
                ", url='" + url + '\'' +
                ", name='" + name + '\'' +
                ", nationality='" + nationality + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        F1Constructor that = (F1Constructor) o;

        return constructorRef.equals(that.constructorRef);

    }

    @Override
    public int hashCode() {
        return constructorRef.hashCode();
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/domain/F1Constructor.kt
package com.gmail.fattazzo.formula1world.domain

import java.io.Serializable

/**
 * @author fattazzo
 *
 *
 * date: 05/06/17
 */
class F1Constructor : Serializable {

    var constructorRef: String? = null

    var url: String? = null

    var name: String? = null

    var nationality: String? = null

    override fun toString(): String {
        return "Constructor{" +
                "constructorRef='" + constructorRef + '\'' +
                ", url='" + url + '\'' +
                ", name='" + name + '\'' +
                ", nationality='" + nationality + '\'' +
                '}'
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as F1Constructor?

        return constructorRef == that!!.constructorRef

    }

    override fun hashCode(): Int {
        return constructorRef!!.hashCode()
    }
}
